File: game/src/main/java/com/scottlogic/hackathon/game/Bot.java
package com.scottlogic.hackathon.game;

import java.util.List;
import java.util.UUID;

public abstract class Bot {
    private final UUID id = UUID.randomUUID();

    public void initialise(final GameState gameState) {
    }

    public abstract List<Move> makeMoves(GameState gameState);

    public UUID getId() {
        return id;
    }

    public abstract  String getDisplayName();
}

File: python-contestant/src/contestant/bot.py
import logging
from typing import List

from bots.base import Bot
from game.move import Move
from game.state import GameState
from game.team import Team

logger = logging.getLogger()


class ExampleBot(Bot):

    def __init__(self, team: Team):
        super().__init__(team)

    def initialise(self, game_state: GameState) -> None:
        pass

    def make_moves(self, game_state: GameState) -> List[Move]:
        logger.info(game_state)
        return []
